apps/explorer/src/pages/transaction-result/TransactionData.tsx

```diff
@@ -0,0 +1,279 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { CoinFormat, useFormatCoin } from '@mysten/core';
+import {
+    getGasData,
+    getTotalGasUsed,
+    getTransactionKind,
+    getTransactionKindName,
+    type ProgrammableTransaction,
+    SUI_TYPE_ARG,
+    type SuiTransactionBlockResponse,
+} from '@mysten/sui.js';
+
+import { InputsCard } from '~/pages/transaction-result/programmable-transaction-view/InputsCard';
+import { TransactionsCard } from '~/pages/transaction-result/programmable-transaction-view/TransactionsCard';
+import { CopyToClipboard } from '~/ui/CopyToClipboard';
+import { DescriptionItem } from '~/ui/DescriptionList';
+import { Divider } from '~/ui/Divider';
+import { Heading } from '~/ui/Heading';
+import { CheckpointSequenceLink, ObjectLink } from '~/ui/InternalLink';
+import { Text } from '~/ui/Text';
+import {
+    TransactionBlockCard,
+    TransactionBlockCardSection,
+} from '~/ui/TransactionBlockCard';
+
+function GasAmount({
+    amount,
+    isHighlighted,
+}: {
+    amount?: bigint | number;
+    isHighlighted?: boolean;
+}) {
+    const [formattedAmount, symbol] = useFormatCoin(
+        amount,
+        SUI_TYPE_ARG,
+        CoinFormat.FULL
+    );
+
+    return (
+        <div className="flex flex-wrap gap-1">
+            <div className="flex flex-wrap items-center gap-1">
+                <Text
+                    variant="pBody/medium"
+                    color={isHighlighted ? 'success-dark' : 'steel-darker'}
+                >
+                    {formattedAmount}
+                </Text>
+                <Text
+                    variant="subtitleSmall/medium"
+                    color={isHighlighted ? 'success-dark' : 'steel-darker'}
+                >
+                    {symbol}
+                </Text>
+            </div>
+
+            <div className="flex flex-wrap items-center text-body font-medium text-steel">
+                ({amount?.toLocaleString()}
```

Comment:
nit: let's use the `Text` component here instead of putting the text styles on the outer `div`

Reply:
That does not work very well since we want `({amount?.toLocaleString()}` and `MIST` to have different stylings. Problem of wrapping `Text` nodes over each other is they just overwriting stylings and it's actually a lot more messy than just 1 off case here